astropy/io/fits/tests/test_connect.py

```diff
@@ -311,3 +311,19 @@ def test_unicode_column(tmpdir):
 
     with pytest.raises(UnicodeEncodeError):
         t2.write(str(tmpdir.join('test.fits')), overwrite=True)
+
+
+def test_unit_warnings_read_write(tmpdir):
+    filename = str(tmpdir.join('test_unit.fits'))
+    t1 = Table([[1, 2], [3, 4]], names=['a', 'b'])
+    t1['a'].unit = 'm/s'
+    t1['b'].unit = 'not-a-unit'
+
+    with catch_warnings() as l:
```

Comment:
I don't like `l` either, but was doing this for consistency with other `catch_warnings` in this test module.  We can blame @astrofrog in 6c0c1a227.  :smile: